passes/opt/opt_rmdff.cc

```diff
@@ -452,6 +459,71 @@ bool handle_dff(RTLIL::Module *mod, RTLIL::Cell *dff)
 		dff->unsetPort("\\E");
 	}
 
+	if (sat && has_init) {
+		bool removed_sigbits = false;
+
+		// Create netlist for the module if not already available
+		if (!netlists.count(mod)) {
+			netlists.emplace(mod, Netlist(mod));
+			comb_filters.emplace(mod, comb_cells_filt());
+		}
+
+		// Load netlist for the module from the pool
+		Netlist &net = netlists.at(mod);
+
+
+		// For each register bit, try to prove that it cannot change from the initial value. If so, remove it
+		for (int position = 0; position < GetSize(sig_d); position += 1) {
+			RTLIL::SigBit q_sigbit = sig_q[position];
+			RTLIL::SigBit d_sigbit = sig_d[position];
+
+			if ((!q_sigbit.wire) || (!d_sigbit.wire)) {
+				continue;
+			}
+
+			ezSatPtr ez;
+			SatGen satgen(ez.get(), &net.sigmap);
```

Comment:
This creates a new SAT instance for each bit of the FF. Generally it is better to re-use SAT instances, at least for the cases where we have reason to assume that the input cone will have large overlap with the previous cone. Therefore we should at least re-use the same sat instance for all bits of the same FF.